From ff72f8350e8b2ded813163150a7d1874061ed589 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 17 Apr 2013 18:06:25 +0200
Subject: [PATCH 04/12] qt_functions: temporary remove isEmpty check

* now we assume that every build will provide QT_EXTERNAL_HOST_BINS value
* isEmpty works correctly only with qmake variables (e.g. $$FOO -
  isEmpty(FOO)), but doesn't work with system properties like $$[FOO].

Upstream-Status: Pending

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 mkspecs/features/qt_functions.prf | 6 +++++-
 mkspecs/features/qt_tool.prf      | 2 +-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/mkspecs/features/qt_functions.prf b/mkspecs/features/qt_functions.prf
index e249960..105ba4c 100644
--- a/mkspecs/features/qt_functions.prf
+++ b/mkspecs/features/qt_functions.prf
@@ -195,7 +195,11 @@ defineTest(qtPrepareTool) {
     $$1 = $$eval(QT_TOOL.$${2}.command)
     isEmpty($$1) {
         $$1 = $$[QT_EXTERNAL_HOST_BINS]/$$2
-        isEmpty($$[QT_EXTERNAL_HOST_BINS]) {
+        # for some reason isEmpty does not work here, FIXME before submitting upstream
+        # DEBUG 1: mkspecs/features/qt_functions.prf:198: calling built-in isEmpty(/OE/oe-core/tmp-eglibc/sysroots/x86_64-linux/usr/bin/qt5)
+        # DEBUG 1: mkspecs/features/qt_functions.prf:198: test function returned true
+        # DEBUG 1: mkspecs/features/qt_functions.prf:198: taking 'then' branch
+        !exists($$[QT_EXTERNAL_HOST_BINS]/$$2) {
             $$1 = $$[QT_HOST_BINS/get]/$$2
         }
         contains(QMAKE_HOST.os, Windows):!contains($$1, .*\\.(exe|bat)$) {
diff --git a/mkspecs/features/qt_tool.prf b/mkspecs/features/qt_tool.prf
index 5518933..b0e4f28 100644
--- a/mkspecs/features/qt_tool.prf
+++ b/mkspecs/features/qt_tool.prf
@@ -31,7 +31,7 @@ load(qt_targets)
 
 # If we are doing a prefix build, create a "module" pri which enables
 # qtPrepareTool() to work with the non-installed build.
-!build_pass:!exists($$[QT_INSTALL_PREFIX]/.qmake.cache):isEmpty($$[QT_EXTERNAL_HOST_BINS]) {
+!build_pass:!exists($$[QT_INSTALL_PREFIX]/.qmake.cache):!exists($$[QT_EXTERNAL_HOST_BINS]) {
 
     isEmpty(MODULE):MODULE = $$TARGET
 
-- 
1.8.2.1

